aspen/src/lib.rs

```diff
@@ -71,33 +71,40 @@ impl Linter {
             .transpose()?;
 
         let injected_trees = self
-            .injection
-            .as_ref()
-            .map(|Injection { query, language }| {
-                // run the injection query with the injection query and the original language
+            .injections
+            .iter()
+            .filter_map(|Injection { query, language }| {
+                // run the injection query with the syntax tree formed from the original language
                 let mut injection_parser = Parser::new();
                 injection_parser.set_language(*language).unwrap();
-                Query::new(self.language, query).map(|query| {
-                    let capture_idx = query.capture_index_for_name("injection.content").unwrap();
-                    QueryCursor::new()
-                        .matches(&query, root_node, src.as_bytes())
-                        .flat_map(|m| m.captures)
-                        .filter(|c| c.index == capture_idx)
-                        .map(|c| c.node.range())
-                        .filter_map(|original_range| {
-                            let (start, end) = (original_range.start_byte, original_range.end_byte);
-                            injection_parser.parse(&src[start..end], None).map(|tree| {
-                                InjectedTree {
-                                    tree,
-                                    original_range,
-                                }
+                Query::new(self.language, query)
+                    .map(|query| {
+                        // TODO: handle this error gracefully
+                        let content_capture = query
+                            .capture_index_for_name("injection.content")
+                            .expect("could not find capture `injection.content`");
```

Comment:
`unwrap_or(-1)` ?

Reply:
and just emit an error that can be caught by sentry or something

Reply:
or if it is serious enough, just bubble an error up.

Reply:
i think analyzer authors should receive this error as soon as they write an injection query that does not contain an `injection.content` capture. which means early exiting when such a query is first read, i think. not sure what a good way to enforce that is, but i am thinking `Injection::new` should panic when such a query is found.

Reply:
hmm, in Java, such things never cause an unrecoverable error at runtime, and instead report an exception to sentry. Tests will throw an assertion exception when such a bad case is found. would that work here?

Reply:
in that case, `Injection::new` should be a `Result<Injection>`, seems like the best option.

Reply:
makes sense, queries can be invalid, and injections can also be invalid due to a bad query or other reasons I can't think of.

Reply:
reworked the error handling a bit, invalid queries are checked for in `Injection::new`, the query is guaranteed to be:

- a valid tree-sitter query
- contain a capture group named `injection.content`, so we can index into it without any worry